libcalico-go/lib/epstatusfile/status_file_writer.go

```diff
@@ -0,0 +1,190 @@
+// Copyright (c) 2025 Tigera, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package epstatusfile
+
+import (
+	"encoding/json"
+	"errors"
+	"io/fs"
+	"os"
+	"path/filepath"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/projectcalico/calico/felix/proto"
+)
+
+const (
+	dirStatus  = "endpoint-status"
+	statusUp   = "up"
+	statusDown = "down"
+)
+
+func GetDirStatus() string {
+	return dirStatus
+}
+
+// EndpointStatusFileWriter writes workload endpoint statuses to the file system.
+type EndpointStatusFileWriter struct {
+	statusDirPrefix string
+
+	Filesys
+}
+
+type Filesys interface {
+	Create(name string) (*os.File, error)
+	Remove(name string) error
+	Mkdir(name string, perm os.FileMode) error
+	ReadDir(name string) ([]os.DirEntry, error)
+	ReadFile(name string) ([]byte, error)
+	WriteFile(name string, data []byte, perm os.FileMode) error
+}
+
+type defaultFilesys struct{}
+
+// Create wraps os.Create.
+func (fs *defaultFilesys) Create(name string) (*os.File, error) {
```

Comment:
I think there's a ready-made FS for this.

Reply:
Are you saying something similar to https://github.com/spf13/afero?

Reply:
Actually we are using our own version of filesystem, because in some test we need to evaluate the timing for file operations. Those functionalities are not provided by an off-the-shelf module.

Reply:
I mean https://pkg.go.dev/os#DirFS `os.DirFS(...)`